### PR TITLE
Add support for organization settings

### PR DESCRIPTION
**Pull request by soerenmartius**

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -61,3 +60,34 @@ resource "github_team_membership" "all" {
...
+  has_repository_projects                                      = try(var.settings.has_repository_projects, true)
+  default_repository_permission                                = try(var.settings.default_repository_permission, "read")
+  members_can_create_repositories                              = try(var.settings.members_can_create_repositories, true)
+  members_can_create_public_repositories                       = try(var.settings.members_can_create_public_repositories, true)
+  members_can_create_private_repositories                      = try(var.settings.members_can_create_private_repositories, true)
+  members_can_create_internal_repositories                     = try(var.settings.members_can_create_internal_repositories, null)
+  members_can_create_pages                                     = try(var.settings.members_can_create_pages, true)
+  members_can_create_public_pages                              = try(var.settings.members_can_create_public_pages, true)
+  members_can_create_private_pages                             = try(var.settings.members_can_create_private_pages, false)
```
Comment (mariux) [inline main.tf:84]: should all be false as we can assume we use IaC for this

```suggestion
  members_can_create_repositories                              = try(var.settings.members_can_create_repositories, false)
  members_can_create_public_repositories                       = try(var.settings.members_can_create_public_repositories, false)
  members_can_create_private_repositories                      = try(var.settings.members_can_create_private_repositories, false)
  members_can_create_internal_repositories                     = try(var.settings.members_can_create_internal_repositories, false)
  members_can_create_pages                                     = try(var.settings.members_can_create_pages, false)
  members_can_create_public_pages                              = try(var.settings.members_can_create_public_pages, false)
  members_can_create_private_pages                             = try(var.settings.members_can_create_private_pages, false)
```

**Review by mariux — CHANGES_REQUESTED** [review]

File: main.tf (hunk 1 above)
Comment (soerenmartius) [reply main.tf:84]: should `members_can_create_internal_repositories` not be set to null? it's available in GH Enterprise only according to the documentation?

Comment (soerenmartius) [reply main.tf]: changed in 84aa087

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,16 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+## [0.9.0]
+
+### BREAKING CHANGES
+
+Updated min GitHub provider version to `v5.3` for adding support for the `github_organization_settings` resource
```
Comment (mariux) [inline CHANGELOG.md:14]:
```suggestion
- Drop support for Terraform GitHub Provider `4.x`
- Require Terraform GitHub Provider version `5.3` and above to support `github_organization_settings` resource
```

File: README.md [hunk 3]
```diff
@@ -12,7 +12,7 @@ A [Terraform] module that acts as a wrapper around the Terraform
 [GitHub provider](https://www.terraform.io/docs/providers/github/index.html) and offers a more convenient and tested way
 to manage GitHub Organizations following best practices.
 
-**_This module supports Terraform v1.x and is compatible with the Official Terraform GitHub Provider v4.x from `integrations/github`._**
+**_This module supports Terraform v1.x and is compatible with the Official Terraform GitHub Provider v5.x from `integrations/github`._**
```
Comment (mariux) [inline README.md:15]: please also update the badge

**Review by mariux — CHANGES_REQUESTED** [review]

File: README.md (hunk 3 above)
Comment (soerenmartius) [reply README.md:15]: resolved

**Review by waxb — APPROVED** [review]
lgtm

**Review by mariux — APPROVED** [review]
lgtm

**Pull request merged**